hudi-utilities/src/main/java/org/apache/hudi/utilities/UtilHelpers.java

```diff
@@ -190,7 +190,9 @@ public static HoodieWriteClient createHoodieClient(JavaSparkContext jsc, String
             .withCompactionStrategy(ReflectionUtils.loadClass(strategy)).build())
         .orElse(HoodieCompactionConfig.newBuilder().withInlineCompaction(false).build());
     HoodieWriteConfig config =
-        HoodieWriteConfig.newBuilder().withPath(basePath).withParallelism(parallelism, parallelism)
+        HoodieWriteConfig.newBuilder().withPath(basePath)
+            .withParallelism(parallelism, parallelism)
+            .withBulkInsertParallelism(parallelism)
```

Comment:
@garyli1019 @vinothchandar I assume this won't have any side effect, will it? Though it is also used by `HoodieCompactor`, the `parallelism` setting should only take effect for one of the preset modes (upsert/insert/bulkinsert).

Reply:
yes. should be safe.. its only used by the write client . good callout.

Reply:
Related: https://issues.apache.org/jira/browse/HUDI-10 How does delta streamer determine parallelism?